pkg/docker/errors.go

```diff
@@ -48,62 +48,62 @@ func NewDockerError(err error) (bacErr bacerrors.Error) {
 	case errdefs.IsNotFound(err):
 		return handleNotFoundError(err)
 	case errdefs.IsConflict(err):
-		return bacerrors.New(err.Error()).
+		return bacerrors.New("%s", err.Error()).
 			WithCode(Conflict).
 			WithHTTPStatusCode(http.StatusConflict)
 	case errdefs.IsUnauthorized(err):
-		return bacerrors.New(err.Error()).
+		return bacerrors.New("%s", err.Error()).
 			WithCode(Unauthorized).
 			WithHTTPStatusCode(http.StatusUnauthorized).
 			WithHint("Ensure you have the necessary permissions and that your credentials are correct. " +
 				"You may need to log in to Docker again.")
 	case errdefs.IsForbidden(err):
-		return bacerrors.New(err.Error()).
+		return bacerrors.New("%s", err.Error()).
 			WithCode(Forbidden).
 			WithHTTPStatusCode(http.StatusForbidden).
 			WithHint(fmt.Sprintf("You don't have permission to perform this action. "+
 				"Supply the node with valid Docker login credentials using the %s and %s environment variables",
 				config_legacy.DockerUsernameEnvVar, config_legacy.DockerPasswordEnvVar))
 	case errdefs.IsDataLoss(err):
-		return bacerrors.New(err.Error()).
+		return bacerrors.New("%s", err.Error()).
```

Comment:
I hate we have to do this. Feel free to skip or disable rules that don't ad much value

Reply:
Since `bacerrors.New` is using `fmt.Sprintf` underneath the hood,  it think it is better to use the formatted string approach since it more idiomatic GO, where it nicely keep the format of the original error message, and nicely deal with nil errors. I did the change to use `"%s", err` , instead of `"%s", err.Error()` to remove repetitive string conversion and use the builtin one in SprintF